Library/Homebrew/dependable.rb

```diff
@@ -3,20 +3,12 @@
 module Dependable
   RESERVED_TAGS = [:build, :optional, :recommended, :run]
 
-  def build?
-    tags.include? :build
-  end
-
-  def optional?
-    tags.include? :optional
-  end
-
-  def recommended?
-    tags.include? :recommended
-  end
+  RESERVED_TAGS.each do |tag|
+    method_name = "#{tag}?"
 
-  def run?
-    tags.include? :run
+    define_method method_name do
+      tags.include? tag
+    end
   end
```

Comment:
The issue I have with these dynamically-generated methods is that it breaks autocomplete; make them less discoverable (no documentation); and prevent any static analysis: you don’t know if a method simply doesn’t exist or will be created in the future.